dimod/discrete/cydiscrete_quadratic_model.pyx

```diff
@@ -83,38 +67,19 @@ cdef class cyDiscreteQuadraticModel:
         cdef Py_ssize_t num_samples = samples.shape[0]
         cdef VarIndex num_variables = samples.shape[1]
 
-        cdef Bias[:] energies = np.zeros(num_samples, dtype=self.dtype)
+        cdef Bias[:] energies = np.empty(num_samples, dtype=self.dtype)
 
         cdef Py_ssize_t si, vi
         cdef CaseIndex cu, case_u, cv, case_v
         cdef VarIndex u, v
-        for si in range(num_samples):  # this could be parallelized
+        for si in range(num_samples):
             for u in range(num_variables):
                 case_u = samples[si, u]
 
                 if case_u >= self.num_cases(u):
                     raise ValueError("invalid case")
 
-                cu = self.case_starts_[u] + case_u
-
-                energies[si] += self.bqm_.get_linear(cu)
-
-                for vi in range(self.adj_[u].size()):
-                    v = self.adj_[u][vi]
-
-                    # we only care about the lower triangle
-                    if v > u:
-                        break
-
-                    case_v = samples[si, v]
-
-                    cv = self.case_starts_[v] + case_v
-
-                    out = self.bqm_.get_quadratic(cu, cv)
-
-                    if out.second:
-                        energies[si] += out.first
-
+        self.dqm_.get_energies(&samples[0,0], num_samples, num_variables, &energies[0])
```

Comment:
This is a templated function, will this work ? I saw some examples which indicate it should work. 
Secondly is it okay to pass the address of the starting point of samples the way I did. I could also pass in samples.data but I was wondering in that case templating may not work since samples.data may be just a void pointer.

Reply:
Honestly I am not sure if it will work. You'll have to try and see!

I am not 100% sure how address of a python object like that works. Cython might be able to handle it, might not.